internal/store/database_roles.go

```diff
@@ -59,40 +60,44 @@ func (s *Store) CreateDatabaseRole(groupID string, role *atlas.CustomDBRole) (*a
 func (s *Store) DeleteDatabaseRole(groupID, roleName string) error {
 	switch s.service {
 	case config.CloudService, config.CloudGovService:
-		_, err := s.client.(*atlas.Client).CustomDBRoles.Delete(s.ctx, groupID, roleName)
+		_, err := s.clientv2.CustomDatabaseRolesApi.DeleteCustomDatabaseRole(s.ctx, groupID, roleName).Execute()
 		return err
 	default:
 		return fmt.Errorf("%w: %s", errUnsupportedService, s.service)
 	}
 }
 
 // DatabaseRoles encapsulate the logic to manage different cloud providers.
-func (s *Store) DatabaseRoles(projectID string, opts *atlas.ListOptions) (*[]atlas.CustomDBRole, error) {
+func (s *Store) DatabaseRoles(projectID string, _ *atlas.ListOptions) ([]atlasv2.CustomDBRole, error) {
```

Comment:
if not needed let's delete it
```suggestion
func (s *Store) DatabaseRoles(projectID string) ([]atlasv2.CustomDBRole, error) {
```